### PR TITLE
EG-1179, Text input

### PR DESCRIPTION
**Pull request by cmrhmrc**
Sorry many files changed. Many changes are as a result of adding the visual stanzas to the PageContext which results in changes to the tests

### CODE REVIEW

File: test/models/ocelot/OcelotPackageSpec.scala [hunk 1]
```diff
@@ -89,6 +89,25 @@ class OcelotPackageSpec extends BaseSpec {
     }
   }
 
+  "Text conversion" must {
+    "recognise a valid text" in {
+      val validNumbers: List[String] = List("Hello", "World", "030")
```
Comment (LesWilson) [inline test/models/ocelot/OcelotPackageSpec.scala:94]: can we change the name of this variable

File: test/views/components/InputSpec.scala [hunk 2]
```diff
@@ -75,14 +75,99 @@ class InputSpec extends WordSpec with Matchers with GuiceOneAppPerSuite {
...
+  "English Text Input component" must {
+
+    trait TextTest extends Test {
+      override val input: TextInput = TextInput(Text(i1), Some(Text(i1Hint)), Seq(h2, h3, h4, para1))
+
+      override val inputWithoutBody: TextInput = TextInput(Text(i1), None, Seq.empty)
+      override val inputWithHintAndNoBody: TextInput = TextInput(Text(i1), Some(Text(i1Hint)), Seq.empty)
+      override val inputWithHintAndErrors: TextInput = TextInput(Text(i1), Some(Text(i1Hint)), Seq(para1), Seq(errorMsg))
+      val textInput: TextInput = models.ui.TextInput(Text(), None, Seq.empty)
+      val textInputPage: FormPage = models.ui.FormPage("/url", textInput)
```
Comment (LesWilson) [inline test/views/components/InputSpec.scala:168]: Do we need this TextInput - can we just use the input defined above?
Can we remove the models.ui. from these definitions

**Review by LesWilson — COMMENTED** [review]
a couple of minor changes.  
Other changes we talked about are more cosmetic and we can discuss more when we have time to tidy things up around PageContext creation to reduce impact of future signature changes

File: test/views/components/InputSpec.scala (hunk 2 above)
Comment (cmrhmrc) [reply test/views/components/InputSpec.scala:168]: They look similar, but the one above uses CurrencyInput as the Input implementation, so itsthe same tests but run through with the TextInput type

File: test/models/ocelot/OcelotPackageSpec.scala (hunk 1 above)
Comment (cmrhmrc) [reply test/models/ocelot/OcelotPackageSpec.scala:94]: will do

File: test/views/components/InputSpec.scala (hunk 2 above)
Comment (LesWilson) [reply test/views/components/InputSpec.scala:168]: Sorry, the picture above doesn't show the input I meant - there is a TextInput defined at the top of this section that is used in tests

Comment (cmrhmrc) [reply test/views/components/InputSpec.scala:168]: The one above is unused, deleted now

**Review by LesWilson — APPROVED** [review]

**Pull request merged**